cmd/ingester/app/consumer/deadlock_detector_test.go

```diff
@@ -112,3 +112,34 @@ func TestGlobalPanic(t *testing.T) {
 	d.start()
 	wg.Wait()
 }
+
+func TestNoGlobalPanicIfDeadlockDetectorDisabled(t *testing.T) {
+	l, _ := zap.NewDevelopment()
+	d := deadlockDetector{
+		metricsFactory: metrics.NewLocalFactory(0),
+		logger:         l,
+		interval:       0,
+		panicFunc: func(partition int32) {
+			t.Errorf("Should not panic when deadlock detector is disabled")
+		},
+	}
+
+	d.start()
```

Comment:
should include a call to close(), to ensure it works as expected when goroutines are not running

Reply:
added calls to close() and fixed blocked channels w/o goroutines with setting "closed" flag on detectors